packages/scene-composer/src/components/panels/scene-components/AnimationComponentEditor.spec.tsx

```diff
@@ -0,0 +1,91 @@
+import React, { useCallback } from 'react';
+import { render, fireEvent, screen } from '@testing-library/react';
+
+import { useStore } from '../../../store';
+import { mockNode, mockProvider } from '../../../../tests/components/panels/scene-components/MockComponents';
+import { generateUUID } from '../../../utils/mathUtils';
+
+import { AnimationComponentEditor } from './AnimationComponentEditor';
+
+jest.mock('react', () => ({
+  ...jest.requireActual('react'),
+  useCallback: jest.fn(),
+}));
+
+const updateComponentInternalFn = jest.fn();
+
+const mockEditorConfig = {
+  valueDataBindingProvider: mockProvider,
+};
+
+const baseState = {
+  updateComponentInternal: updateComponentInternalFn,
+  getEditorConfig: jest.fn(() => {
+    return mockEditorConfig;
+  }),
+};
+
+jest.mock('../../../../src/store/Store', () => {
+  const originalModule = jest.requireActual('../../../../src/store/Store');
+  return {
+    __esModule: true,
+    ...originalModule,
+    useSceneDocument: jest.fn(() => ({})),
+  };
+});
+
+describe('AnimationComponentEditor', () => {
+  const mockComponentEmpty: any = {
+    ref: generateUUID(),
+    currentAnimations: [],
+  };
+  const mockComponent: any = {
+    ref: generateUUID(),
+    selector: 2,
+    currentAnimations: ['Action.018', 'Action.013'],
+  };
+
+  const mockComponentExtraPanel: any = {
+    ref: generateUUID(),
+    selector: 3,
+    currentAnimations: ['Action.018', 'Action.013'],
+  };
+  const callbacks = [] as any[];
+  beforeEach(() => {
+    jest.clearAllMocks();
+    (useCallback as jest.Mock).mockImplementation((callback, dep) => {
+      callbacks.push(callback);
+      return callback;
+    });
+    // Get references to specific callbacks in tests by:
+  });
+
+  it('should update the correct animations', () => {
+    const mockComponent: any = {
+      ref: generateUUID(),
+      selector: 2,
+      currentAnimations: ['Action.018', 'Action.013'],
+    };
+    const [updateAnimations, addAnimation] = callbacks;
+    const onUpdate = jest.fn();
+    render(<AnimationComponentEditor node={mockNode} component={mockComponent} onUpdate={onUpdate} />);
+    callbacks[0]('Action.014', 0);
+    callbacks[1]('Action.015');
+  });
+
+  it('should render as expected', () => {
+    useStore('default').setState(baseState);
+    const onUpdate = jest.fn();
+    render(<AnimationComponentEditor node={mockNode} component={mockComponentEmpty} onUpdate={onUpdate} />);
+  });
+
+  it('should render as expected', () => {
```

Comment:
The description matches the test above. make it phrase clear so we can tell the differences in the test cases